thinc/backends/_custom_kernels.py

```diff
@@ -363,3 +409,56 @@ def hash(ids, seed, out=None, threads_per_block=128, num_blocks=128):
         (out, ids, out_size, in_size, ids.shape[0], seed),
     )
     return out
+
+
+def _check_compatible(out, shape: Optional[Tuple] = None):
+    assert out.dtype == "float32", "CUDA kernel can only handle float32"
+    if shape is not None and out.shape != shape:
+        msg = f"array has incorrect shape, expected: {shape}, was: {out.shape}"
+        raise ValueError(msg)
+
+
+def _check_lengths(lengths, n_elems: int):
+    assert lengths.dtype == "int32", "lengths should be encoded as 32-bit integers"
+    if not cupy.all(lengths >= 0):
+        raise ValueError("all sequence lengths must be >= 0")
+    if cupy.sum(lengths) != n_elems:
+        raise IndexError("lengths must sum up to the batch size")
+
+
+def _check_which_maxout(which, B: int, I: int, P: int):
+    shape = (B, I)
+    assert (
+        which.dtype == "int32"
+    ), "maximum index (which) should be encoded as 32-bit integers"
```

Comment:
This would probably be a bit more readible if it uses a `msg` variable like the `if` on the next line does -> then the `assert` can go on one line.

Reply:
Fixed